app/bundles/PageBundle/Entity/VideoHit.php

```diff
@@ -0,0 +1,808 @@
+<?php
+/**
+ * @package     Mautic
+ * @copyright   2014 Mautic Contributors. All rights reserved.
+ * @author      Mautic
+ * @link        http://mautic.org
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\PageBundle\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+use Mautic\CoreBundle\Doctrine\Mapping\ClassMetadataBuilder;
+use Mautic\EmailBundle\Entity\Email;
+use Mautic\LeadBundle\Entity\Lead;
+use Mautic\ApiBundle\Serializer\Driver\ApiMetadataDriver;
+
+/**
+ * Class VideoHit
+ *
+ * @package Mautic\PageBundle\Entity
+ */
+class VideoHit
+{
+    /**
+     * @var int
+     */
+    private $id;
+
+    /**
+     * @var string
+     */
+    private $guid;
+
+    /**
+     * @var \DateTime
+     */
+    private $dateHit;
+
+    /**
+     * @var \DateTime
+     */
+    private $dateLeft;
+
+    /**
+     * @var int
+     */
+    private $timeWatched;
+
+    /**
+     * @var int
+     */
+    private $duration;
+
+    /**
+     * @var Redirect
+     */
+    private $redirect;
+
+    /**
+     * @var \Mautic\LeadBundle\Entity\Lead
+     */
+    private $lead;
+
+    /**
+     * @var \Mautic\CoreBundle\Entity\IpAddress
+     */
+    private $ipAddress;
+
+    /**
+     * @var string
+     */
+    private $country;
+
+    /**
+     * @var string
+     */
+    private $region;
+
+    /**
+     * @var string
+     */
+    private $city;
+
+    /**
+     * @var string
+     */
+    private $isp;
+
+    /**
+     * @var string
+     */
+    private $organization;
+
+    /**
+     * @var integer
+     */
+    private $code;
+
+    /**
+     * @var
+     */
+    private $referer;
+
+    /**
+     * @var
+     */
+    private $url;
+
+    /**
+     * @var string
+     */
+    private $userAgent;
+
+    /**
+     * @var string
+     */
+    private $remoteHost;
+
+    /**
+     * @var string
+     */
+    private $pageLanguage;
+
+    /**
+     * @var string
+     */
+    private $browserLanguages = array();
+
+    /**
+     * @var string
+     */
+    private $source;
```

Comment:
Thoughts on terming these channel and channel id instead of source? Source is remnants of page hits from before we had the concept of channel and channel ID. Thinking it would be good if we keep code going forward synchronous with "channel"